app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java

```diff
@@ -62,8 +65,30 @@ private void initListView() {
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         DividerItemDecoration itemDecor = new DividerItemDecoration(recyclerView.getContext(), DividerItemDecoration.VERTICAL);
         recyclerView.addItemDecoration(itemDecor);
-        addNotifications();
+        if (!NetworkUtils.isInternetConnectionEstablished(this)) {
```

Comment:
IMO this can be replaced by `refresh()` itself.

Reply:
Yup you are correct. I will make the changes.